### ISSUE TITLE
is it possible to run validation with a costum dataset using coco's pretrained model? where the class order is different and random.

### ISSUE THREAD

**Issue by Mirshal** [ultralytics/ultralytics#7733]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have a problem related to validation using the costum dataset that I have with the pretrained coco model. where the dataset class that I have consists of 7 classes with the order ["vehicle", "motorcycle", "public transport", "becak", "truck", "person", "traffic sign"]. 

While the pretrained coco model consists of 80 classes in the order ['person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus', 'train', 'truck', 'boat', 'traffic light', 'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow','elephant', 'bear', 'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie', 'suitcase', 'frisbee','skis', 'snowboard', 'sports ball', 'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard', 'tennis racket', 'bottle', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple', 'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza', 'donut', 'cake', 'chair', 'couch', 'potted plant', 'bed', 'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone', 'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear', 'hair drier', 'toothbrush' ]

how to solve the difference, because the results of precision, recall, and map are very low. 
I would be very grateful if you can help solve this problem.
![image](https://github.com/ultralytics/ultralytics/assets/101452195/e1221a42-e991-40e2-938f-1668d817093c)

**Comment by glenn-jocher** [comment]
@Mirshal yes, it's possible to validate a custom dataset with a different class order using a pretrained COCO model. However, you must ensure that your dataset's annotation files correctly map your classes to the corresponding COCO classes. If there are classes in your dataset that don't exist in COCO, they should be ignored during validation or you should train a new model on your custom dataset with the correct class labels.

Low precision, recall, and mAP scores may indicate that the class mappings are incorrect or that the model is not well-suited to your specific classes. Consider fine-tuning the pretrained model on your dataset to improve performance. For more guidance on training and validation with custom datasets, please refer to our documentation on the Train and Val modes. 📘🚀
